appcontext/app_context.go

```diff
@@ -14,20 +14,25 @@ const (
 	TenantIDKey ContextKey = "tenant_id"
 )
 
-// AppContext carries the context of the current execution.
-type AppContext struct {
-	// Context for cancellation
-	ctx context.Context
+type Context interface {
```

Comment:
I don't get why you need an interface here ?

I mean you should explain it

Reply:
because context.Context is an interface

Reply:
To be honest, it does need to be an interface. I can return the struct directly.

Reply:
The other option is to add a static method where you must pass the context every time. 

```
appcontext.SetUser(parentCtx, "userId")

userId, ok := appcontext.GetUser(parentCtx)
``

Reply:
It's what we did at work